man/sd_event_source_set_ratelimit.xml

```diff
@@ -0,0 +1,150 @@
+    and the <constant>SD_EVENT_OFF</constant> parameter. However it is disabled only temporarily – once the
+    specified interval is over regular operation resumes — until the specified rate limiting is hit the next
+    time. If either the interval or the burst value are specified as zero, rate limiting is turned off. By
+    default event sources do not have rate limiting enabled. Note that rate limiting and disabling via
+    <function>sd_event_source_set_enabled()</function> are independent of each other, and an event source
+    will only effect event loop wake-ups and is dispatched while it both is enabled and rate limiting is not
+    in effect.</para>
+
+    <para><function>sd_event_source_get_ratelimit()</function> may be used to query the current rate limiting
+    parameters set on the event source object <parameter>source</parameter>. The previously set interval and
+    burst vales are returned in the second and third argument.</para>
+
+    <para><function>sd_event_source_is_ratelimited()</function> may be used to query whether the event source
+    is currently affected by rate limiting, i.e. it has recently hit the rate limit and is currently
+    temporarily disabled due to that.</para>
+
+    <para>Rate limiting is currently implemented for I/O, timer, signal, defer and inotify event
+    sources.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Return Value</title>
+
+    <para>On success, <function>sd_event_source_set_ratelimit()</function> and
+    <function>sd_event_source_get_ratelimit()</function> return a non-negative integer. On failure, they
+    return a negative errno-style error code. <function>sd_event_source_is_ratelimited</function> returns
+    zero if rate limiting is currently not in effect and greater than zero if it is in effect; it returns a
+    negative errno-style error code on failure.</para>
+
+    <refsect2>
+      <title>Errors</title>
+
+      <para>Returned errors may indicate the following problems:</para>
+
+      <variablelist>
+        <varlistentry>
+          <term><constant>-EINVAL</constant></term>
+
+          <listitem><para><parameter>source</parameter> is not a valid pointer to an
+          <structname>sd_event_source</structname> object.
+          </para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><constant>-ECHILD</constant></term>
+
+          <listitem><para>The event loop has been created in a different process.</para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><constant>-EDOM</constant></term>
+
+          <listitem><para>It was attempted to use the rate limiting feature on an event source type that does not support rate limiting.</para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><constant>-ENOEXEC</constant></term>
+
+          <listitem><para>It was attempted to read the current rate limiting parameters with
```

Comment:
`<function>sd_event_source_get_ratelimit()</function> or <function>sd_event_source_is_ratelimited()</function> was called on a event source that doesn't have rate limiting configured.`
(But note that I asked for the interface to be changed, so this description might be out of date.)